dali/operators/generic/reduce/histogram.h

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_REDUCE_HISTOGRAM_H_
+#define DALI_OPERATORS_GENERIC_REDUCE_HISTOGRAM_H_
+
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/generic/reduce/axes_helper.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/pipeline/operator/operator.h"
+
+#include "opencv2/imgproc.hpp"
+
+namespace dali {
+namespace hist_detail {
+
+inline bool is_simple_reduction(const std::vector<int> &reduction_axes, int ndims,
+                                int hist_dim = 1) {
+  int dim = ndims - hist_dim;
+
+  DALI_ENFORCE(dim >= 1);
```

Comment:
I don't think simple queries like that should throw. Passing `hist_dim < 1` constitutes a coding error. Since this function is internal, `assert` should be enough.